### PR TITLE
[New Rules] PowerShell Suspicious Payload Encoded and Compressed

### PR DESCRIPTION
**Pull request by w0rk3r**
## Summary

Identifies the use of decompression and base64 decoding combined in PowerShell scripts, which Malware and Hacktools
heavily use to deobfuscate payloads and load them directly in memory to bypass defenses.

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/windows/defense_evasion_posh_compressed.toml [hunk 1]
```diff
@@ -0,0 +1,68 @@
+[metadata]
+creation_date = "2021/10/19"
+maturity = "production"
+updated_date = "2021/10/19"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies the use of decompression and base64 decoding combined in PowerShell scripts, which Malware and Hacktools
```
Comment (brokensound77) [inline rules/windows/defense_evasion_posh_compressed.toml:9]:
```suggestion
Identifies the use of .Net functionality for decompression and base64 decoding combined in PowerShell scripts, which Malware and security tools
```

will need to re-lint this for length

Comment (brokensound77) [reply rules/windows/defense_evasion_posh_compressed.toml:9]: I think there are several other rules which rely on calls to .Net classes and functionality which we do not mention in the description which could use similar language.

File: rules/windows/defense_evasion_posh_compressed.toml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+heavily use to deobfuscate payloads and load them directly in memory to bypass defenses.
+"""
+from = "now-9m"
```
Comment (brokensound77) [inline rules/windows/defense_evasion_posh_compressed.toml:12]: This could be a noisy one for prod tools and so could use an FP entry

**Review by brokensound77 — APPROVED** [review]
Few comments and LGTM

File: rules/windows/defense_evasion_posh_compressed.toml (hunk 2 above)
Comment (w0rk3r) [reply rules/windows/defense_evasion_posh_compressed.toml:13]: Done

**Review by Samirbous — APPROVED** [review]

**Comment by baserock** [comment]
This rule has a bunch of false positives for citrix config and microsoft AAD verified scripts that use base64 encoded certificates.

**Comment by w0rk3r** [comment]
Hey @baserock, can you open an issue sharing some details on this? I'll be more than happy to take a look on tuning opportunities

**Comment by w0rk3r** [comment]
@baserock I've taken a look at the queries, and it seems that the field type change (https://github.com/elastic/integrations/issues/1776) caused our queries to behave a little differently and cause partial matches. I've pushed a PR (https://github.com/elastic/detection-rules/pull/1807) that updates this query. It would be great if you could test and provide feedback

**Pull request merged**